pkg/test/clusterdeployment/clusterdeployment.go

```diff
@@ -136,6 +136,8 @@ func WithClusterPoolReference(namespace, poolName, claimName string) Option {
 			PoolName:  poolName,
 			ClaimName: claimName,
 		}
+		now := metav1.Now()
+		clusterDeployment.Status.ClaimedTimestamp = &now
```

Comment:
NTS: This was the fun test bug where the extra (Status) Update() to [sync this](https://github.com/openshift/hive/pull/1528/files#diff-bbf4d3cb24b8cac87951b84ca973747902c00d4a9d662648c8e760a37569581dR625) would cause the fake client to actually delete a CD with its DeletionTimestamp set.

Having it set on a claimed CD is The Right Thing™ in any case -- we just happened to have a test case that made sure we did it, in a roundabout way.